src/native/eventpipe/ep.c

```diff
@@ -289,13 +291,15 @@ ep_provider_callback_data_init_copy (
 	EP_ASSERT (provider_callback_data_src != NULL);
 
 	*provider_callback_data_dst = *provider_callback_data_src;
+	provider_callback_data_dst->filter_data = ep_rt_utf8_string_dup (provider_callback_data_src->filter_data);
 	return provider_callback_data_dst;
 }
 
 void
 ep_provider_callback_data_fini (EventPipeProviderCallbackData *provider_callback_data)
 {
-	;
+	ep_return_void_if_nok (provider_callback_data != NULL);
```

Comment:
You nee to call fini from ep_provider_callback_data_free as well or this will leak in case you do alloc/free.